controllers/util/provider.go

```diff
@@ -64,16 +76,52 @@ func GenerateOnboardingToken(tokenLifetimeInHours int, privateKeyPath string, st
 	return fmt.Sprintf("%s.%s", encodedPayload, encodedSignature), nil
 }
 
-func readAndDecodePrivateKey(privateKeyPath string) (*rsa.PrivateKey, error) {
-	pemString, err := os.ReadFile(privateKeyPath)
+func readAndDecodePrivateKey() (*rsa.PrivateKey, error) {
+	cl, err := newClient()
 	if err != nil {
-		return nil, fmt.Errorf("failed to read private key: %v", err)
+		klog.Exitf("failed to create client: %v", err)
+	}
+	ctx := context.Background()
+	operatorNamespace, err := GetOperatorNamespace()
+	if err != nil {
+		klog.Exitf("unable to get operator namespace: %v", err)
 	}
 
-	Block, _ := pem.Decode(pemString)
+	privateSecret := &corev1.Secret{}
+	privateSecret.Name = onboardingValidationPrivateKeySecretName
+	privateSecret.Namespace = operatorNamespace
+	err1 := cl.Get(ctx, types.NamespacedName{Name: onboardingValidationPrivateKeySecretName, Namespace: operatorNamespace}, privateSecret)
+	if err1 != nil {
+		return nil, fmt.Errorf("failed to get private secret: %v", err1)
+	}
```

Comment:
1. There should be no need to create a new client. In most cases this code is being used there is already a client that is used on the call sire of `GenerateOnboardingToken`. It is important to use that client instead of the new client for many reasons, some of which are:
- The client is a cached client and the object is already loaded
- The client is a mock/testing client used for testing and does not execute to a valid k8s API server

2. I am not sure you are taking the right approach by trying to load the secret here. This code path was and should be a compute-only code path, this means it will benefit us to not introduce any k8s-specific concerns into it.  My approach will be to load the secret outside of the `GenerateOnboardingToken` util and provide the private key payload an argument

Reply:
1) Used existing client on the call site on GenerateOnboardingToken, 

2) The secrets are loaded in a separate function and the private key is decoded and passed to the GenerateOnboardingToken